src/main/java/org/waarp/openr66/protocol/localhandler/TransferActions.java

```diff
@@ -483,6 +484,22 @@ public void request(LocalChannel channel, RequestPacket packet)
         }
         session.setReady(true);
         Configuration.configuration.getLocalTransaction().setFromId(runner, localChannelReference);
+
+        // Set read/write limit
+        long remoteLimit = packet.getLimit();
+        long localLimit = localChannelReference.getChannelLimit(
+                runner.isSender());
+        // Take the minimum speed
+        logger.trace("Received limit {}", packet.getLimit());
+        logger.trace("Local limit {}", localChannelReference.getChannelLimit(runner.isSender()));
+        if (localLimit <= 0) {
+            localLimit = remoteLimit;
+        } else if (remoteLimit > 0 && remoteLimit < localLimit) {
+            localLimit = remoteLimit;
+        }
+        localChannelReference.setChannelLimit(runner.isSender(), localLimit);
+        packet.setLimit(localLimit);
+
```

Comment:
Just a quick question: did you manage a client in an older version, such that limit is not sent, and therefore not in the RequestPacket ? What is the behavior (on both side, even, server older than client) ?

Reply:
The Limit field is carried on the Json part of the RequestPacket.
Older R66 partner not using Json Formating of the packet are not concerned by this change.
Older R66 partner using Json will not parse the Limit field and send a limitless RequestPacket.

(The Json formating of the RequestPacket make it the easiest place to implement changes without breaking retro-compatibility)

Reply:
Yes, I know the parameter will be either absent or ignored, according to the version, but my question was more on the "procedure" part: is this taking into account such parameter being absent (so 0) on client or server side being the new version (and therefore taking into consideration this _missing_ parameter) ?

Reply:
Test have been made between old version and this one  : it works well for get and put.